File: app/src/theme/tokens.ts
/**
 * Cappy Design Tokens
 *
 * Direct translation of the cappy.css design system into TypeScript.
 * The source of truth lives in /handoff/cappy-web/cappy.css. When that
 * file changes, mirror the change here.
 *
 * Tokens are organized as:
 *   - palette: raw color ramps (teal-500, blue-500, etc.)
 *   - semantic: meaning-bearing tokens (bg, fg, brand, focus)
 *   - dose: the dose-safety status system (due/early/recent/overdue)
 *   - typography: font families and scale
 *   - spacing, radii, motion: layout primitives
 *
 * Light and dark variants live side by side. The ThemeProvider picks one.
 */

// ─────────────────────────────────────────────────────────────────────────
// PALETTE — raw color ramps. Same in light and dark.
// ─────────────────────────────────────────────────────────────────────────

export const palette = {
  teal: {
    50: '#E8F6F2',
    100: '#C9EBE1',
    200: '#9FDDCC',
    300: '#6FCBB4',
    400: '#3FB89C',
    500: '#18A78D',
    600: '#128873',
    700: '#0E6D5C',
    800: '#0B5346',
    900: '#083E34',
    950: '#052822',
  },
  blue: {
    50: '#EAF2FB',
    100: '#CFE2F6',
    200: '#A6C9EE',
    300: '#6FA8E0',
    400: '#3E86D2',
    500: '#1E6FC4',
    600: '#155AA6',
    700: '#114887',
    800: '#0E3A6E',
    900: '#0B2D55',
  },
  tan: { 100: '#F2E7D6', 300: '#DCC39B', 500: '#C29A66', 700: '#8F6E43' },
  mint: { 50: '#F1F9F6', 100: '#E8F5F1', 200: '#D6EFE8', 300: '#BFE4D8', 400: '#9ED3C2' },
  sage: { 100: '#DDF1E6', 300: '#8DD2AE', 500: '#4FB088', 600: '#2E9E6E', 700: '#237A56' },
  amber: { 100: '#FCEBD2', 300: '#F2C173', 500: '#E89B2D', 600: '#D97A0E', 700: '#B05E08' },
  coral: { 100: '#FBE3E1', 300: '#F2A8A2', 500: '#E36B62', 600: '#D84A4A', 700: '#B43838' },
  slate: {
    0: '#FFFFFF',
    25: '#FBFAF7',
    50: '#F7F5F0',
    100: '#EFEBE2',
    200: '#DDD8CC',
    300: '#BDB8AC',
    400: '#8F8B82',
    500: '#6A6760',
    600: '#4A4844',
    700: '#322F2C',
    800: '#1F1D1B',
    900: '#131211',
  },
  cream: '#FBF8F2',
  cream2: '#F4EFE5',
} as const;

// ─────────────────────────────────────────────────────────────────────────
// SPACING / RADII / MOTION — same for both themes
// ─────────────────────────────────────────────────────────────────────────

export const spacing = {
  xs: 4, // --space-1
  sm: 8, // --space-2
  md: 12, // --space-3
  base: 16, // --space-4
  lg: 20, // --space-5
  xl: 24, // --space-6
  xxl: 32, // --space-8
  xxxl: 40, // --space-10
  huge: 48, // --space-12
  giant: 64, // --space-16
  // Screen-edge gutter (matches --screen-gutter)
  gutter: 20,
  // Min touch target (matches --tap-min)
  tapMin: 44,
  // Tab bar height (matches --tabbar-h)
  tabBar: 84,
} as const;

export const radii = {
  xs: 4, // --radius-1
  sm: 8, // --radius-2
  md: 12, // --radius-3
  base: 16, // --radius-4
  lg: 24, // --radius-5
  pill: 999, // --radius-pill
  sheet: 28, // --sheet-radius
} as const;

export const motion = {
  fast: 200, // --motion-fast
  base: 320, // --motion-base
  slow: 480, // --motion-slow
} as const;

// ─────────────────────────────────────────────────────────────────────────
// TYPOGRAPHY
// ─────────────────────────────────────────────────────────────────────────

/**
 * NOTE: React Native uses string font family names that match the loaded
 * font file. We load Inter, NunitoSans, DM Mono, and Baloo 2 via
 * expo-font at app startup. The string here is what RN expects after
 * the font is loaded.
 */
export const fonts = {
  display: 'NunitoSans-Bold', // headings / numerals
  displaySemibold: 'NunitoSans-SemiBold',
  sans: 'Inter-Regular', // UI body
  sansMedium: 'Inter-Medium',
  sansSemibold: 'Inter-SemiBold',
  sansBold: 'Inter-Bold',
  mono: 'DMMono-Regular', // units (mL, mg, °F)
  monoMedium: 'DMMono-Medium',
  brand: 'Baloo2-SemiBold', // "Cappy!" wordmark only
  brandBold: 'Baloo2-Bold',
} as const;

export const fontSize = {
  xs: 12,
  sm: 14,
  smPlus: 15,
  base: 16,
  lg: 18,
  xl: 20,
  xxl: 24,
  xxxl: 28,
  display: 32,
  displayLg: 40,
  doseNumeral: 48,
} as const;

export const lineHeight = {
  xs: 16,
  sm: 20,
  base: 22,
  lg: 26,
  xl: 28,
  xxl: 32,
  display: 36,
  doseNumeral: 48,
} as const;

// ─────────────────────────────────────────────────────────────────────────
// SEMANTIC TOKENS — light and dark
// ─────────────────────────────────────────────────────────────────────────

export type SemanticTokens = {
  // Backgrounds
  bg: string;
  bgMuted: string;
  bgCard: string;
  bgInset: string;
  bgTint: string;
  bgTint2: string;
  // Foreground / text
  fg: string;
  fg1: string;
  fg2: string;
  fg3: string;
  fgMuted: string;
  fgOnBrand: string;
  // Lines
  border: string;
  borderStrong: string;
  hairline: string;
  // Brand & accents
  brand: string;
  brandHover: string;
  brandPress: string;
  brandTint: string;
  link: string;
  // Capybara-blue accent (used for "log dose" CTA)
  accent2: string;
  accent2Hover: string;
  accent2Press: string;
  accent2Tint: string;
  accent2Fg: string;
  // Status
  success: string;
  warn: string;
  error: string;
  // Focus
  focus: string;
  focusHalo: string;
  // Dose-safety palette
  doseDueSolid: string;
  doseDueFg: string;
  doseDueBg: string;
  doseDueRing: string;
  doseEarlySolid: string;
  doseEarlyFg: string;
  doseEarlyBg: string;
  doseEarlyRing: string;
  doseRecentSolid: string;
  doseRecentFg: string;
  doseRecentBg: string;
  doseRecentRing: string;
  doseOverdueSolid: string;
  doseOverdueFg: string;
  doseOverdueBg: string;
  doseOverdueRing: string;
  // NFC
  nfcCore: string;
  nfcGlow: string;
  nfcRing: string;
  // Scrim / sheet
  scrim: string;
  sheetGrabber: string;
  // Shadow color (RN uses shadowColor + opacity)
  shadow: string;
};

export const lightTokens: SemanticTokens = {
  bg: palette.cream,
  bgMuted: palette.cream2,
  bgCard: palette.slate[0],
  bgInset: palette.slate[50],
  bgTint: palette.mint[100],
  bgTint2: palette.mint[200],
  fg: palette.slate[800],
  fg1: palette.slate[800],
  fg2: palette.slate[600],
  fg3: palette.slate[500],
  fgMuted: palette.slate[400],
  fgOnBrand: '#FFFFFF',
  border: 'rgba(11,30,29,0.08)',
  borderStrong: 'rgba(11,30,29,0.14)',
  hairline: 'rgba(11,30,29,0.06)',
  brand: palette.teal[500],
  brandHover: palette.teal[600],
  brandPress: palette.teal[700],
  brandTint: palette.teal[50],
  link: palette.teal[600],
  accent2: palette.blue[500],
  accent2Hover: palette.blue[600],
  accent2Press: palette.blue[700],
  accent2Tint: palette.blue[50],
  accent2Fg: '#FFFFFF',
  success: palette.sage[600],
  warn: palette.amber[600],
  error: palette.coral[600],
  focus: palette.teal[500],
  focusHalo: 'rgba(24,167,141,0.22)',
  // Dose status
  doseDueSolid: palette.teal[500],
  doseDueFg: palette.teal[800],
  doseDueBg: palette.teal[50],
  doseDueRing: 'rgba(24,167,141,0.30)',
  doseEarlySolid: palette.amber[600],
  doseEarlyFg: palette.amber[700],
  doseEarlyBg: palette.amber[100],
  doseEarlyRing: 'rgba(217,122,14,0.28)',
  doseRecentSolid: palette.blue[500],
  doseRecentFg: palette.blue[700],
  doseRecentBg: palette.blue[50],
  doseRecentRing: 'rgba(30,111,196,0.26)',
  doseOverdueSolid: palette.coral[600],
  doseOverdueFg: palette.coral[700],
  doseOverdueBg: palette.coral[100],
  doseOverdueRing: 'rgba(216,74,74,0.28)',
  // NFC
  nfcCore: palette.blue[500],
  nfcGlow: 'rgba(30,111,196,0.22)',
  nfcRing: 'rgba(30,111,196,0.40)',
  scrim: 'rgba(8,26,24,0.42)',
  sheetGrabber: 'rgba(11,30,29,0.18)',
  shadow: '#0B1E1D',
};

export const darkTokens: SemanticTokens = {
  bg: '#0B1717',
  bgMuted: '#0F1C1B',
  bgCard: '#13201F',
  bgInset: '#1A2A29',
  bgTint: '#14302C',
  bgTint2: '#1B3F39',
  fg: '#ECECE6',
  fg1: '#ECECE6',
  fg2: '#B5B7B0',
  fg3: '#8A8C85',
  fgMuted: '#5E605A',
  fgOnBrand: '#FFFFFF',
  border: 'rgba(255,255,255,0.08)',
  borderStrong: 'rgba(255,255,255,0.16)',
  hairline: 'rgba(255,255,255,0.05)',
  brand: palette.teal[400],
  brandHover: palette.teal[300],
  brandPress: palette.teal[200],
  brandTint: 'rgba(54,169,154,0.16)',
  link: palette.teal[300],
  accent2: '#4D93DD',
  accent2Hover: '#6FA8E0',
  accent2Press: palette.blue[200],
  accent2Tint: 'rgba(77,147,221,0.16)',
  accent2Fg: '#06121F',
  success: palette.sage[500],
  warn: palette.amber[500],
  error: palette.coral[500],
  focus: palette.teal[300],
  focusHalo: 'rgba(111,195,179,0.25)',
  doseDueSolid: palette.teal[400],
  doseDueFg: '#8FE6CF',
  doseDueBg: 'rgba(24,167,141,0.16)',
  doseDueRing: 'rgba(63,184,156,0.34)',
  doseEarlySolid: palette.amber[500],
  doseEarlyFg: '#F4C173',
  doseEarlyBg: 'rgba(217,122,14,0.18)',
  doseEarlyRing: 'rgba(232,155,45,0.30)',
  doseRecentSolid: '#4D93DD',
  doseRecentFg: palette.blue[200],
  doseRecentBg: 'rgba(77,147,221,0.16)',
  doseRecentRing: 'rgba(77,147,221,0.32)',
  doseOverdueSolid: palette.coral[500],
  doseOverdueFg: palette.coral[300],
  doseOverdueBg: 'rgba(216,74,74,0.18)',
  doseOverdueRing: 'rgba(227,107,98,0.32)',
  nfcCore: '#4D93DD',
  nfcGlow: 'rgba(77,147,221,0.26)',
  nfcRing: 'rgba(77,147,221,0.46)',
  scrim: 'rgba(0,0,0,0.58)',
  sheetGrabber: 'rgba(255,255,255,0.22)',
  shadow: '#000000',
};

// Shadow presets (RN-specific shape; iOS uses these directly, Android uses elevation)
export const shadows = (color: string) => ({
  shadow1: {
    shadowColor: color,
    shadowOffset: { width: 0, height: 1 },
    shadowOpacity: 0.04,
    shadowRadius: 4,
    elevation: 1,
  },
  shadow2: {
    shadowColor: color,
    shadowOffset: { width: 0, height: 4 },
    shadowOpacity: 0.08,
    shadowRadius: 12,
    elevation: 3,
  },
  shadow3: {
    shadowColor: color,
    shadowOffset: { width: 0, height: 12 },
    shadowOpacity: 0.16,
    shadowRadius: 24,
    elevation: 8,
  },
});

// Convenience type for the full theme passed through context
export type Theme = {
  mode: 'light' | 'dark';
  tokens: SemanticTokens;
  palette: typeof palette;
  spacing: typeof spacing;
  radii: typeof radii;
  motion: typeof motion;
  fonts: typeof fonts;
  fontSize: typeof fontSize;
  lineHeight: typeof lineHeight;
  shadows: ReturnType<typeof shadows>;
};

File: ios-native/Cappy/DesignSystem/Tokens.swift
//
//  Tokens.swift
//  Cappy
//
//  Direct Swift port of the Cappy design tokens (app/src/theme/tokens.ts).
//  When the source design system changes, mirror the change here.
//
//  Organized as:
//    - Palette:  raw color ramps (teal-500, blue-500, …)
//    - Semantic: meaning-bearing tokens (bg, fg, brand, dose-status, …)
//    - Spacing / Radii / Motion / Type scale: layout primitives
//
//  Light and dark variants live side by side; `Theme` picks one.
//

import SwiftUI

// MARK: - Palette (raw ramps — identical in light & dark)

enum Palette {
    // Teal — brand
    static let teal50 = Color(cappy: "#E8F6F2")
    static let teal100 = Color(cappy: "#C9EBE1")
    static let teal200 = Color(cappy: "#9FDDCC")
    static let teal300 = Color(cappy: "#6FCBB4")
    static let teal400 = Color(cappy: "#3FB89C")
    static let teal500 = Color(cappy: "#18A78D")
    static let teal600 = Color(cappy: "#128873")
    static let teal700 = Color(cappy: "#0E6D5C")
    static let teal800 = Color(cappy: "#0B5346")
    static let teal900 = Color(cappy: "#083E34")
    static let teal950 = Color(cappy: "#052822")

    // Blue — "log dose" accent
    static let blue50 = Color(cappy: "#EAF2FB")
    static let blue100 = Color(cappy: "#CFE2F6")
    static let blue200 = Color(cappy: "#A6C9EE")
    static let blue300 = Color(cappy: "#6FA8E0")
    static let blue400 = Color(cappy: "#3E86D2")
    static let blue500 = Color(cappy: "#1E6FC4")
    static let blue600 = Color(cappy: "#155AA6")
    static let blue700 = Color(cappy: "#114887")
    static let blue800 = Color(cappy: "#0E3A6E")
    static let blue900 = Color(cappy: "#0B2D55")

    static let mint50 = Color(cappy: "#F1F9F6")
    static let mint100 = Color(cappy: "#E8F5F1")
    static let mint200 = Color(cappy: "#D6EFE8")
    static let mint300 = Color(cappy: "#BFE4D8")
    static let mint400 = Color(cappy: "#9ED3C2")

    static let sage100 = Color(cappy: "#DDF1E6")
    static let sage300 = Color(cappy: "#8DD2AE")
    static let sage500 = Color(cappy: "#4FB088")
    static let sage600 = Color(cappy: "#2E9E6E")
    static let sage700 = Color(cappy: "#237A56")

    static let amber100 = Color(cappy: "#FCEBD2")
    static let amber300 = Color(cappy: "#F2C173")
    static let amber500 = Color(cappy: "#E89B2D")
    static let amber600 = Color(cappy: "#D97A0E")
    static let amber700 = Color(cappy: "#B05E08")

    static let coral100 = Color(cappy: "#FBE3E1")
    static let coral300 = Color(cappy: "#F2A8A2")
    static let coral500 = Color(cappy: "#E36B62")
    static let coral600 = Color(cappy: "#D84A4A")
    static let coral700 = Color(cappy: "#B43838")

    static let slate0 = Color(cappy: "#FFFFFF")
    static let slate25 = Color(cappy: "#FBFAF7")
    static let slate50 = Color(cappy: "#F7F5F0")
    static let slate100 = Color(cappy: "#EFEBE2")
    static let slate200 = Color(cappy: "#DDD8CC")
    static let slate300 = Color(cappy: "#BDB8AC")
    static let slate400 = Color(cappy: "#8F8B82")
    static let slate500 = Color(cappy: "#6A6760")
    static let slate600 = Color(cappy: "#4A4844")
    static let slate700 = Color(cappy: "#322F2C")
    static let slate800 = Color(cappy: "#1F1D1B")
    static let slate900 = Color(cappy: "#131211")

    static let cream = Color(cappy: "#FBF8F2")
    static let cream2 = Color(cappy: "#F4EFE5")
}

// MARK: - Spacing / Radii / Motion

enum Space {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let base: CGFloat = 16
    static let lg: CGFloat = 20
    static let xl: CGFloat = 24
    static let xxl: CGFloat = 32
    static let xxxl: CGFloat = 40
    static let huge: CGFloat = 48
    static let giant: CGFloat = 64
    static let gutter: CGFloat = 20
    static let tapMin: CGFloat = 44
    static let tabBar: CGFloat = 84
}

enum Radius {
    static let xs: CGFloat = 4
    static let sm: CGFloat = 8
    static let md: CGFloat = 12
    static let base: CGFloat = 16
    static let lg: CGFloat = 24
    static let pill: CGFloat = 999
    static let sheet: CGFloat = 28
}

enum Motion {
    static let fast: Double = 0.20
    static let base: Double = 0.32
    static let slow: Double = 0.48
}

// MARK: - Type scale

enum FontSizeToken {
    static let xs: CGFloat = 12
    static let sm: CGFloat = 14
    static let smPlus: CGFloat = 15
    static let base: CGFloat = 16
    static let lg: CGFloat = 18
    static let xl: CGFloat = 20
    static let xxl: CGFloat = 24
    static let xxxl: CGFloat = 28
    static let display: CGFloat = 32
    static let displayLg: CGFloat = 40
    static let doseNumeral: CGFloat = 48
}

// MARK: - Semantic tokens

/// Meaning-bearing color tokens. Two instances exist: `.light` and `.dark`.
struct SemanticTokens {
    // Backgrounds
    var bg: Color
    var bgMuted: Color
    var bgCard: Color
    var bgInset: Color
    var bgTint: Color
    var bgTint2: Color
    // Foreground
    var fg: Color
    var fg1: Color
    var fg2: Color
    var fg3: Color
    var fgMuted: Color
    var fgOnBrand: Color
    // Lines
    var border: Color
    var borderStrong: Color
    var hairline: Color
    // Brand
    var brand: Color
    var brandHover: Color
    var brandPress: Color
    var brandTint: Color
    var link: Color
    // Capybara-blue accent (the "log dose" CTA)
    var accent2: Color
    var accent2Hover: Color
    var accent2Press: Color
    var accent2Tint: Color
    var accent2Fg: Color
    // Status
    var success: Color
    var warn: Color
    var error: Color
    // Focus
    var focus: Color
    var focusHalo: Color
    // Dose-safety palette
    var doseDueSolid: Color
    var doseDueFg: Color
    var doseDueBg: Color
    var doseDueRing: Color
    var doseEarlySolid: Color
    var doseEarlyFg: Color
    var doseEarlyBg: Color
    var doseEarlyRing: Color
    var doseRecentSolid: Color
    var doseRecentFg: Color
    var doseRecentBg: Color
    var doseRecentRing: Color
    var doseOverdueSolid: Color
    var doseOverdueFg: Color
    var doseOverdueBg: Color
    var doseOverdueRing: Color
    // NFC
    var nfcCore: Color
    var nfcGlow: Color
    var nfcRing: Color
    // Scrim / sheet
    var scrim: Color
    var sheetGrabber: Color
    // Shadow
    var shadow: Color
}

extension SemanticTokens {
    static let light = SemanticTokens(
        bg: Palette.cream,
        bgMuted: Palette.cream2,
        bgCard: Palette.slate0,
        bgInset: Palette.slate50,
        bgTint: Palette.mint100,
        bgTint2: Palette.mint200,
        fg: Palette.slate800,
        fg1: Palette.slate800,
        fg2: Palette.slate600,
        fg3: Palette.slate500,
        fgMuted: Palette.slate400,
        fgOnBrand: Color.white,
        border: Color(cappy: "rgba(11,30,29,0.08)"),
        borderStrong: Color(cappy: "rgba(11,30,29,0.14)"),
        hairline: Color(cappy: "rgba(11,30,29,0.06)"),
        brand: Palette.teal500,
        brandHover: Palette.teal600,
        brandPress: Palette.teal700,
        brandTint: Palette.teal50,
        link: Palette.teal600,
        accent2: Palette.blue500,
        accent2Hover: Palette.blue600,
        accent2Press: Palette.blue700,
        accent2Tint: Palette.blue50,
        accent2Fg: Color.white,
        success: Palette.sage600,
        warn: Palette.amber600,
        error: Palette.coral600,
        focus: Palette.teal500,
        focusHalo: Color(cappy: "rgba(24,167,141,0.22)"),
        doseDueSolid: Palette.teal500,
        doseDueFg: Palette.teal800,
        doseDueBg: Palette.teal50,
        doseDueRing: Color(cappy: "rgba(24,167,141,0.30)"),
        doseEarlySolid: Palette.amber600,
        doseEarlyFg: Palette.amber700,
        doseEarlyBg: Palette.amber100,
        doseEarlyRing: Color(cappy: "rgba(217,122,14,0.28)"),
        doseRecentSolid: Palette.blue500,
        doseRecentFg: Palette.blue700,
        doseRecentBg: Palette.blue50,
        doseRecentRing: Color(cappy: "rgba(30,111,196,0.26)"),
        doseOverdueSolid: Palette.coral600,
        doseOverdueFg: Palette.coral700,
        doseOverdueBg: Palette.coral100,
        doseOverdueRing: Color(cappy: "rgba(216,74,74,0.28)"),
        nfcCore: Palette.blue500,
        nfcGlow: Color(cappy: "rgba(30,111,196,0.22)"),
        nfcRing: Color(cappy: "rgba(30,111,196,0.40)"),
        scrim: Color(cappy: "rgba(8,26,24,0.42)"),
        sheetGrabber: Color(cappy: "rgba(11,30,29,0.18)"),
        shadow: Color(cappy: "#0B1E1D")
    )

    static let dark = SemanticTokens(
        bg: Color(cappy: "#0B1717"),
        bgMuted: Color(cappy: "#0F1C1B"),
        bgCard: Color(cappy: "#13201F"),
        bgInset: Color(cappy: "#1A2A29"),
        bgTint: Color(cappy: "#14302C"),
        bgTint2: Color(cappy: "#1B3F39"),
        fg: Color(cappy: "#ECECE6"),
        fg1: Color(cappy: "#ECECE6"),
        fg2: Color(cappy: "#B5B7B0"),
        fg3: Color(cappy: "#8A8C85"),
        fgMuted: Color(cappy: "#5E605A"),
        fgOnBrand: Color.white,
        border: Color(cappy: "rgba(255,255,255,0.08)"),
        borderStrong: Color(cappy: "rgba(255,255,255,0.16)"),
        hairline: Color(cappy: "rgba(255,255,255,0.05)"),
        brand: Palette.teal400,
        brandHover: Palette.teal300,
        brandPress: Palette.teal200,
        brandTint: Color(cappy: "rgba(54,169,154,0.16)"),
        link: Palette.teal300,
        accent2: Color(cappy: "#4D93DD"),
        accent2Hover: Color(cappy: "#6FA8E0"),
        accent2Press: Palette.blue200,
        accent2Tint: Color(cappy: "rgba(77,147,221,0.16)"),
        accent2Fg: Color(cappy: "#06121F"),
        success: Palette.sage500,
        warn: Palette.amber500,
        error: Palette.coral500,
        focus: Palette.teal300,
        focusHalo: Color(cappy: "rgba(111,195,179,0.25)"),
        doseDueSolid: Palette.teal400,
        doseDueFg: Color(cappy: "#8FE6CF"),
        doseDueBg: Color(cappy: "rgba(24,167,141,0.16)"),
        doseDueRing: Color(cappy: "rgba(63,184,156,0.34)"),
        doseEarlySolid: Palette.amber500,
        doseEarlyFg: Color(cappy: "#F4C173"),
        doseEarlyBg: Color(cappy: "rgba(217,122,14,0.18)"),
        doseEarlyRing: Color(cappy: "rgba(232,155,45,0.30)"),
        doseRecentSolid: Color(cappy: "#4D93DD"),
        doseRecentFg: Palette.blue200,
        doseRecentBg: Color(cappy: "rgba(77,147,221,0.16)"),
        doseRecentRing: Color(cappy: "rgba(77,147,221,0.32)"),
        doseOverdueSolid: Palette.coral500,
        doseOverdueFg: Palette.coral300,
        doseOverdueBg: Color(cappy: "rgba(216,74,74,0.18)"),
        doseOverdueRing: Color(cappy: "rgba(227,107,98,0.32)"),
        nfcCore: Color(cappy: "#4D93DD"),
        nfcGlow: Color(cappy: "rgba(77,147,221,0.26)"),
        nfcRing: Color(cappy: "rgba(77,147,221,0.46)"),
        scrim: Color(cappy: "rgba(0,0,0,0.58)"),
        sheetGrabber: Color(cappy: "rgba(255,255,255,0.22)"),
        shadow: Color.black
    )
}
